### PR TITLE
feat(lsp): add first-class LSP server support to install pipeline 

### PR DESCRIPTION
**Pull request by stbenjam**
## Description

Implements Language Server Protocol (LSP) integration mirroring the MCP pattern:

- LSPDependency model: Parse and validate LSP server configs from apm.yml and plugin.json
- Plugin parser: Extract inline lspServers from plugin.json and .lsp.json files, convert to dependencies.lsp entries
- LSPIntegrator: Orchestrates transitive collection, deduplication, installation, stale cleanup, and lockfile persistence
- Lockfile: Add lsp_servers and lsp_configs fields for drift detection
- Install pipeline: Wire LSP integration after MCP integration
- InstallContext: Add direct_lsp_deps field for LSP deps tracking

LSP config is written to .lsp.json at project root (Claude Code auto-discovers this file). Unlike MCP which targets multiple runtimes, LSP is Claude Code-only for now.

This enables plugins that declare both MCP servers and LSP servers (like gopls for Go support) to have both integrated end-to-end.

## Type of change

- [ ] Bug fix
- [X] New feature
- [ ] Documentation
- [ ] Maintenance / refactor

## Testing

- [X] Tested locally
- [X] All existing tests pass
- [X] Added tests for new functionality (if applicable)

### CODE REVIEW

**Comment by sergio-sisternes-epam** [comment]
@stbenjam thank you for this contribution — LSP server support is a great addition to the install pipeline.

We've confirmed that `.lsp.json` is an established Anthropic standard for Claude Code plugins (documented in the official plugin docs, and actively used by both official and third-party marketplace plugins), so this is well-motivated and aligned with upstream.

During review we identified a few quality gaps that we're fixing on your branch to help move this forward:

- Adding comprehensive unit tests (the PR currently has zero test coverage for 1200+ lines of new code)
- Adding a CHANGELOG entry
- Verifying the `LSPIntegrator` follows our `BaseIntegrator` pattern conventions
- Fixing any lint issues

We'll push these directly to your branch to expedite things. Deferring the final merge decision to @danielmeppiel.

Thanks again for picking this up — the MCP-mirroring approach and the docs are solid work.

**Comment by danielmeppiel** [comment]
## APM Review Panel: `ship_with_followups`

> PR #1424 adds first-class LSP server support on the accepted design path and is internally consistent after the shepherd fold pass.

cc @stbenjam @sergio-sisternes-epam -- a fresh advisory pass is ready for your review.

The panel weighed the maintainer-approved LSP dependency-kind design as settled and focused on implementation quality, schema and lockfile shape, docs, tests, and install UX. The fold pass addressed the in-scope findings: LSP config parsing now preserves valid zero values, validates path-shaped and typed fields, forwards LSP logger output, includes LSP counts in install summaries, registers the docs page in navigation, and adds lockfile regression coverage. The remaining README growth suggestion is not folded because this repo's docs rule requires explicit maintainer approval before README edits.

**Aligned with:** portability by manifest: LSP servers are now represented in `apm.yml` and preserved in `apm.lock.yaml`; pragmatic as npm: install output now names LSP server counts; oss community driven: the contributor feature is credited in the changelog and docs.

### Panel summary

| Persona | B | R | N | Takeaway |
|---|---|---|---|---|
| Python Architect | 0 | 0 | 0 | Falsy-value and validation concerns were folded; architecture now mirrors MCP consistently. |
| CLI Logging Expert | 0 | 0 | 0 | LSP install output now flows through the logger and final summary. |
| DevX UX Expert | 0 | 0 | 0 | LSP install behavior is more visible and less tightly named to the MCP gate. |
| Supply Chain Security Expert | 0 | 0 | 0 | Path-shaped LSP workspace config and extension maps are validated before serialization. |
| OSS Growth Hacker | 0 | 1 | 0 | README top-of-funnel mention remains a separate maintainer-approved follow-up. |
| Auth Expert | 0 | 0 | 0 | No auth behavior, token resolution, or credential flow changed. |
| Doc Writer | 0 | 0 | 0 | LSP docs are reachable from the sidebar and consumer overview. |
| Test Coverage Expert | 0 | 0 | 0 | LSP lockfile round-trip and semantic-equivalence regression traps were added. |

> B = highest-severity findings, R = recommended, N = nits.
> Counts are signal strength, not gates. The maintainer ships.

### Top 1 follow-up

1. **[OSS Growth Hacker]** Consider adding a README hero example for `dependencies.lsp` after maintainer approval -- the README is the top-of-funnel surface, but repo rules require explicit approval before changing it.

### Recommendation

Ship after maintainer review. The accepted LSP dependency-kind contract is now documented, tested, surfaced in install output, and represented in lockfile drift checks. Track the README growth item separately because it crosses the repository's explicit README-approval boundary.

---

### Reservations carried from strategic-alignment

- Large accepted LSP dependency-kind change; verify implementation consistency, tests, docs, schema, and lockfile shape -- addressed by `b8945af3b848793718f9ba0940100ee33b9dddae`.
- PR was conflicting/dirty with failing checks -- addressed by rebasing onto `origin/main` and pushing `b8945af3b848793718f9ba0940100ee33b9dddae`.

### Folded in this run

- (panel) Preserve valid zero-valued LSP config fields and validate `workspaceFolder` plus `extensionToLanguage` values -- resolved in `b8945af3b848793718f9ba0940100ee33b9dddae`.
- (panel) Forward the install logger through LSP integration and include LSP server counts in post-install summaries -- resolved in `b8945af3b848793718f9ba0940100ee33b9dddae`.
- (panel) Add LSP lockfile round-trip and semantic-equivalence regression tests -- resolved in `b8945af3b848793718f9ba0940100ee33b9dddae`.
- (panel) Register the LSP docs page in the Starlight sidebar and consumer overview, and align lockfile docs with persisted `name` fields -- resolved in `b8945af3b848793718f9ba0940100ee33b9dddae`.
- (panel) Rewrite the changelog line as a user-facing LSP management feature note -- resolved in `b8945af3b848793718f9ba0940100ee33b9dddae`.

### Deferred (out-of-scope follow-ups)

- (panel) Add `dependencies.lsp` to the README hero example -- scope boundary: repository docs rules require explicit maintainer approval before README edits; suggested follow-up: maintainer-approved README growth update.

### Regression-trap evidence (mutation-break gate)

- `tests/unit/deps/test_lsp_dependency.py::TestFromDict::test_camel_case_zero_values_preserved` -- deleted `_first_defined` non-None guard; test FAILED as expected; guard restored.
- `tests/unit/deps/test_lsp_dependency.py::TestValidateEdgeCases::test_workspace_folder_traversal_raises` -- disabled `workspaceFolder` traversal validation; test FAILED as expected; guard restored.
- `tests/unit/deps/test_lsp_dependency.py::TestValidateEdgeCases::test_extension_to_language_values_must_be_strings` -- disabled string-map validation; test FAILED as expected; guard restored.
- `tests/test_lockfile.py::TestLockFile::test_lsp_servers_round_trip` -- deleted `lsp_servers` serialization; test FAILED as expected; guard restored.
- `tests/test_lockfile.py::TestLockFile::test_lsp_configs_round_trip` -- deleted `lsp_configs` serialization; test FAILED as expected; guard restored.
- `tests/test_lockfile.py::TestLockFileSemanticEquivalence::test_changed_lsp_servers_not_equivalent` -- deleted `lsp_servers` equivalence comparison; test FAILED as expected; guard restored.
- `tests/test_lockfile.py::TestLockFileSemanticEquivalence::test_changed_lsp_configs_not_equivalent` -- deleted `lsp_configs` equivalence comparison; test FAILED as expected; guard restored.

### Lint contract

`uv run --extra dev ruff check src/ tests/` and `uv run --extra dev ruff format --check src/ tests/` both silent before push.

### CI

`gh pr checks 1424 --repo microsoft/apm` reports all available checks successful on `b8945af3b848793718f9ba0940100ee33b9dddae`: 1 successful (`license/cla`), 0 pending, 0 failing (after 0 CI fix iteration(s)).

### Mergeability status

Captured from `gh pr view 1424 --json mergeable,mergeStateStatus,statusCheckRollup` immediately after the last push of this run.

| PR | head SHA | CEO stance | iters | folds | defers | Copilot rounds | CI | mergeable | mergeStateStatus | notes |
|----|----------|------------|-------|-------|--------|----------------|----|-----------|------------------|-------|
| #1424 | `b8945af` | ship_with_followups | 1 | 5 | 1 | 2 | green | MERGEABLE | BLOCKED | awaiting maintainer review |

### Convergence

1 outer iteration; 2 Copilot rounds. Final panel stance: `ship_with_followups`.

Ready for maintainer review.

---

<details>
<summary>Full per-persona findings considered</summary>

- Python Architect: folded valid-zero parsing and validation concerns.
- CLI Logging Expert: folded logger forwarding and LSP count visibility.
- DevX UX Expert: folded install-output recovery hint and separated the LSP gate name from the MCP gate in code.
- Supply Chain Security Expert: folded path-shaped workspace validation and extension map type validation.
- OSS Growth Hacker: folded changelog polish; deferred README hero update for approval.
- Auth Expert: inactive; no auth-relevant behavior changed.
- Doc Writer: folded sidebar/consumer index registration and lockfile example accuracy.
- Test Coverage Expert: folded lockfile round-trip and semantic-equivalence tests.

</details>

<sub>This panel is advisory. It does not block merge. Re-apply the `panel-review` label after addressing feedback to re-run.</sub>

File: docs/src/content/docs/consumer/index.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ You're here because you want to install someone else's APM packages and use them
 | You need org-private packages or your own marketplace                          | [Private and org packages](./private-and-org-packages/) |
 | You manage `apm.yml` and need lockfile / dependency commands                   | [Manage dependencies](./manage-dependencies/)           |
 | You want APM to wire MCP servers (GitHub, Atlassian, ...) into your tools      | [Install MCP servers](./install-mcp-servers/)           |
+| You want APM to wire LSP servers into Claude Code                               | [Install LSP servers](./install-lsp-servers/)           |
```
Comment (danielmeppiel) [inline docs/src/content/docs/consumer/index.md]: Copilot CLI also supports it, we have to maintain vendor neutrality - https://docs.github.com/en/copilot/concepts/agents/copilot-cli/lsp-servers

File: docs/src/content/docs/consumer/install-lsp-servers.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  order: 5
+---
+
+`apm install` handles three dependency kinds: APM packages
+(see [Install Packages](../install-packages/)), MCP servers
+(see [Install MCP servers](../install-mcp-servers/)), and LSP servers.
+This page covers LSP servers: how you declare them, what gets written,
+and how the install pipeline manages their lifecycle.
+
+LSP integration currently targets **Claude Code only**. The dependency
```
Comment (danielmeppiel) [inline docs/src/content/docs/consumer/install-lsp-servers.md]: add support for Copilot CLI https://docs.github.com/en/copilot/concepts/agents/copilot-cli/lsp-servers

**Review by danielmeppiel — CHANGES_REQUESTED** [review]
add support for copilot cli, and keep vendor neutrality in docs and implementation, wire through integrators and target mechanics https://docs.github.com/en/copilot/concepts/agents/copilot-cli/lsp-servers
